tests/AppBundle/Controller/MembershipControllerTest.php

```diff
@@ -283,12 +279,86 @@ public function testPinInterestsPersistsInterestsForNonActivatedAdherent()
 
-        /* @var Adherent $adherent */
+        /** @var Adherent $adherent */
         $adherent = $this->getAdherentRepository()->findByEmail('michelle.dufour@example.ch');
 
         $this->assertSame(array_values($chosenInterests), $adherent->getInterests());
     }
 
+    public function testChooseNearbyCommitteeWithoutNewAdherentId()
+    {
+        $this->client->request(Request::METHOD_GET, '/inscription/choisir-des-comites');
+
+        $this->assertResponseStatusCode(Response::HTTP_NOT_FOUND, $this->client->getResponse());
+    }
+
+    public function testChooseNearbyCommitteeWithWrongNewAdherentId()
+    {
+        $this->client->getContainer()->get('session')->set(MembershipUtils::NEW_ADHERENT_ID, 'wrong id');
+        $this->client->request(Request::METHOD_GET, '/inscription/choisir-des-comites');
+
+        $this->assertResponseStatusCode(Response::HTTP_NOT_FOUND, $this->client->getResponse());
+    }
+
+    public function testChooseNearbyCommittee()
+    {
+        $this->client->getContainer()->get('session')->set(MembershipUtils::NEW_ADHERENT_ID, 1);
+
+        $crawler = $this->client->request(Request::METHOD_GET, '/inscription/choisir-des-comites');
+
+        $boxPattern = '#app_membership_choose_nearby_committee_committees > div';
+
+        $this->assertResponseStatusCode(Response::HTTP_OK, $this->client->getResponse());
+        $this->assertCount(3, $boxes = $crawler->filter($boxPattern));
+
+        /** @var Committee[] $committees */
+        $committees = $this->getCommitteeRepository()->findBy([], ['id' => 'desc'], 3);
+
+        foreach ($boxes as $i => $box) {
+            $checkbox = $crawler->filter($boxPattern.' input[type="checkbox"][name="app_membership_choose_nearby_committee[committees][]"]');
+
+            $this->assertSame((string) $committees[$i]->getUuid(), $checkbox->eq($i)->attr('value'));
+            $this->assertSame($committees[$i]->getName(), $crawler->filter($boxPattern.' h5')->eq($i)->text());
+        }
+    }
+
+    public function testChooseNearbyCommitteePersistsMembershipForNonActivatedAdherent()
+    {
+        /** @var Adherent $adherent */
+        $adherent = $this->getAdherentRepository()->findByEmail('michelle.dufour@example.ch');
+
+        /** @var CommitteeMembership[] $memberships */
+        $memberships = $this->getCommitteeMembershipRepository()->findBy(['adherentUuid' => (string) $adherent->getUuid()]);
+
+        $this->assertFalse($adherent->isEnabled());
+        $this->assertCount(0, $memberships);
+
+        $this->client->getContainer()->get('session')->set(MembershipUtils::NEW_ADHERENT_ID, $adherent->getId());
+
+        $crawler = $this->client->request(Request::METHOD_GET, '/inscription/choisir-des-comites');
+
+        $this->assertResponseStatusCode(Response::HTTP_OK, $this->client->getResponse());
+
+        /** @var Committee[] $committees */
+        $committees = $this->getCommitteeRepository()->findBy([], ['id' => 'desc'], 3);
```

Comment:
For this one, this will be the case in the next issue. Just for test purpose here.